packages/playwright-core/src/server/injected/recorder/recorder.ts

```diff
@@ -271,6 +275,15 @@ class RecordActionTool implements RecorderTool {
       return;
     }
 
+    if (isRangeInput(target)) {
+      this._recorder.delegate.recordAction?.({
+        name: 'fill',
+        selector: this._hoveredModel!.selector,
+        signals: [],
+        text: target.value,
+      });
+    }
```

Comment:
It definitely needs a `return` there. It only records one instruction but it's just because it's not commiting it, but it is calling the `recordAction` function twice, yes.

I'll do that change and rebase it.